### PR TITLE
Fixing css issues, adding srcset + new pager

### PR DESCRIPTION
**Pull request by angelorepole**

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -40,54 +36,48 @@
...
+  padding: 10px 0 0 10px;
+  
   text-align: left;
   font-family: "Neutra2Display-Medium";
-  line-height: 27px
+  font-size: 21px;
 }
 
-@media screen and (max-width: 500px) {
+/*@media screen and (max-width: 500px) {
```
Comment (keithamus) [inline index.css]: Rather than commenting out, it's better to just remove. We have git if we need to come back to some old code.

File: index.css [hunk 2]
```diff
@@ -114,33 +104,94 @@
...
+  width: auto;
+}
+.Pager-previous svg,
+.Pager-next svg {
+  width: 60px;
+  height: 60px;
+  fill: #dadada;
+}
+.Pager-previous:hover svg,
+.Pager-next:hover svg {
```
Comment (keithamus) [inline index.css]: These selectors should be prefixed with the component name, if we have another Pager component on the page (that's not part of the Gallery), then it'll be styled with these, which isn't ideal.

**Comment by angelorepole** [comment]
Thanks. All fixed please re-review

File: index.es6 [hunk 3]
```diff
@@ -108,10 +113,12 @@ export default class Gallery extends React.Component {
         <div className="Gallery--features">
           {captionDiv}
           <div className="Gallery--features-scenechange">
-            <SceneChanger
+            <Pager
               sceneTotal={sceneTotal}
               defaultSceneIndex={sceneIndex}
-              onChangeIndex={this.passState.bind(this)}/>
+              onChangeIndex={this.passState.bind(this)}
+              prevNext="arrows"
```
Comment (keithamus) [inline index.es6]: This is no longer correct now, is it?

Comment (angelorepole) [inline index.es6]: Correct, however I can't test the change as (please advise):

every time I npm link a branch into another I get  throw new Error('No element indexed by ' + aIdx);
         ^
Error: No element indexed by 1

angelorepole [11:05 AM]
what am I doing wrong?

angelorepole [11:08 AM]
the only thing I can see is that in the node_modules folder in @economist of the parent component (eg: component-gallery) the linked repo (eg: component-pager) has all the .es6 files while other repos (eg: component-imagecaption) just have the index.js file and no .es6 files. Is there a relation?

Comment (keithamus) [inline index.es6]: It appears to be something to do with the inline source maps that babel uses. If you recompile without the `--source-maps=inline` it should work.

**Comment by angelorepole** [comment]
All fixed please re-review

File: index.es6 [hunk 4]
```diff
@@ -63,6 +61,10 @@ export default class Gallery extends React.Component {
     }, 550);
   }
 
+  getSrcSet(image) {
+    return Object.keys(image).map((key) => `${image[key]} ${key}`).join(',');
+  }
```
Comment (keithamus) [inline index.es6]: If srcset is empty (e.g. its an svg in the content json) then this will crash the gallery. Suggest wrapping an if.

**Comment by angelorepole** [comment]
Please re-review (this goes with the pager branch in review https://github.com/the-economist-editorial/component-pager/pull/1)

File: index.es6 [hunk 5]
```diff
@@ -30,12 +28,12 @@ export default class Gallery extends React.Component {
...
       sceneIndex: props.defaultSceneIndex,
       hideImage: false,
-      showPager: false,
+      showPager: '',
     };
   }
 
   componentDidMount() {
-    this.setState({ showPager: 'Gallery-loaded' });
+    this.setState({ showPager: ' Gallery-loaded' });
```
Comment (keithamus) [inline index.es6]: I'm confused about this. Why do we only want to show the pager when the component is mounted? Surely we always want it shown.

Comment (angelorepole) [inline index.es6]: I guess the idea is that the pager is driven by click events only available when the DOM is fully loaded and javascript is available.

Comment (keithamus) [inline index.es6]: They can be rendered without click handlers being bound though. React will ignore click handlers for `React.renderToString` but it is still useful to have the buttons rendered upfront.

Comment (angelorepole) [inline index.es6]: Can you elaborate on that please? This give us a non-javascript version of the gallery, although not a gallery anymore but more of an image with caption.

Comment (keithamus) [inline index.es6]: Ahh yes, I see. Good point, I retract my comments then :smile:

**Pull request closed without merging**